src/App.tsx

```diff
@@ -15,36 +17,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  None = '',
+  Alphabetically = 'sortAlphabetically',
+  ByLength = 'sortLength',
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [reversed, setReversed] = useState(false);
+
+  const visibleGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case 'sortAlphabetically':
+        return good1.localeCompare(good2);
+      case 'sortLength':
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    visibleGoods.reverse();
+  }
+
+  const resetGoods = () => {
+    setSortField(SortType.None);
+    setReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={
+            sortField === 'sortAlphabetically'
+              ? 'button is-info'
+              : 'button is-info is-light'
+          }
+          onClick={() => setSortField(SortType.Alphabetically)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={
+            sortField === 'sortLength'
+              ? 'button is-success'
+              : 'button is-success is-light'
+          }
+          onClick={() => setSortField(SortType.ByLength)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={
+            reversed ? 'button is-warning' : 'button is-warning is-light'
+          }
+          onClick={() => setReversed(!reversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(reversed || sortField) && (
```

Comment:
The condition `(reversed || sortField)` should be adjusted to check for `SortType.None` instead of a falsy check on `sortField`. Use `sortField !== SortType.None` to accurately determine if sorting is applied.